### PR TITLE
ci: update to go 1.16 and fix go mod

### PR DESCRIPTION
**Pull request by mattklein123**
1) Update to go 1.16
2) Fix go mod issues
3) Switch to GH actions and build docker image inline
4) Other small fixes

Signed-off-by: Matt Klein <mklein@lyft.com>

### CODE REVIEW

**Comment by mattklein123** [comment]
@snowp @kyessenov this should be ready for review. Once we review and merge this I will remove the circleci requirements from the build. I will follow up later with the TODO fixes.

File: Makefile [hunk 1]
```diff
@@ -12,17 +12,19 @@ PKG 		:= github.com/envoyproxy/go-control-plane
...
 .PHONY: cover
 cover:
 	@build/coverage.sh
 
-.PHONY: format
-format:
-	@goimports -local $(PKG) -w -l pkg
+.PHONY: check_format
+check_format:
+	@gofmt -l $(shell go list -f '{{.Dir}}' ./...) | tee /dev/stderr | read && echo "Files failed gofmt" && exit 1 || true
```
Comment (snowp) [inline Makefile:27]: Should we still be using goimports? iiuc it does what go fmt does but more

**Review by snowp — COMMENTED** [review]
Thanks, this looks right to me. Seems like a nice cleanup!

Just one question

File: Makefile (hunk 1 above)
Comment (mattklein123) [reply Makefile:27]: I see let me check that. It didn't seem like goimports worked in the new container so I just made this what ratelimit was doing.

**Comment by mattklein123** [comment]
@snowp OK I think this should be back to goimports

File: Makefile [hunk 2]
```diff
@@ -12,9 +12,11 @@ PKG 		:= github.com/envoyproxy/go-control-plane
 build:
 	@go build ./pkg/... ./envoy/...
 
+# TODO(mattklein123): See the note in TestLinearConcurrentSetWatch() for why we set -parallel here
+# This should be
```
Comment (snowp) [inline Makefile:16]: Partial comment

**Review by snowp — APPROVED** [review]
Thanks this looks good just the one comment nit. 

Might be worth moving towards using https://golangci-lint.run/ to mange running linters etc., but we can tackle that separately.

**Comment by mattklein123** [comment]
@snowp updated

**Review by snowp — APPROVED** [review]
Thanks!

**Pull request merged**